dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/AbstractSnowflakeConnector.java

```diff
@@ -117,8 +117,12 @@ private DataSource createUserPasswordDataSource(@Nonnull ConnectorArguments argu
       throws SQLException {
     Driver driver =
         newDriver(arguments.getDriverPaths(), "net.snowflake.client.jdbc.SnowflakeDriver");
-    String password = arguments.getPasswordOrPrompt();
-    return new SimpleDriverDataSource(driver, url, arguments.getUser(), password);
+    Properties prop = new Properties();
+
+    prop.put("user", arguments.getUser());
+    prop.put("password", arguments.getPasswordOrPrompt());
+    prop.put("authenticator", "username_password_mfa");
```

Comment:
Can we add some test? 

I'm wondering about maintaining.  Let's say in the future some one will just delete `runtimeOnly libs.jna` because it's not used explicitly. How can we prevent this situation?

Reply:
Good idea, thanks! I added a simple unit test and a comment.